anxcloud/resource_ip_address.go

```diff
@@ -139,6 +267,24 @@ func resourceIPAddressDelete(ctx context.Context, d *schema.ResourceData, m inte
 	c := m.(providerContext).legacyClient
 	a := address.NewAPI(c)
 
+	if addr, err := a.Get(ctx, d.Id()); handleNotFoundError(err) != nil {
+		// return unhandled error
+		return diag.FromErr(err)
+	} else if err != nil {
+		// handle not found error by just deleting the resource
+		d.SetId("")
+		return nil
```

Comment:
Uh, just from reading the source code, I think that the error handling is wrong here. If I read it right, the first part checks the not found error and the second the generic one. Right now the reaction is vice-versa, so I think the code was intended to look like this:

```suggestion
	if addr, err := a.Get(ctx, d.Id()); handleNotFoundError(err) != nil {
		// handle not found error by just deleting the resource
		d.SetId("")
		return nil
	} else if err != nil {
		// return unhandled error
		return diag.FromErr(err)
```

Reply:
I think the original code was correct. But to reduce any confusion, I've introduced a new `isLegacyNotFoundError` helper.